gwdetchar/lasso/tests/test_main.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for gwdetchar.lasso.__main__
+"""
+
+import tempfile
+
+import numpy as np
+
+from gwpy.timeseries import TimeSeries
+
+from .. import __main__ as lasso
+
+
+__author__ = 'Michael Lowry <michaeljohn.lowry@ligo.org>'
+
+
+# global test objects
+
+# volatile temp file to write expected timeseries to
+TEMP = tempfile.NamedTemporaryFile()
+TEST_CHANNEL = 'L1:DMT-SNSW_EFFECTIVE_RANGE_MPC.mean'
+TEST_START = 0
+TEST_END = TEST_START+1000
+TEST_ARRAY = np.linspace(TEST_START, TEST_END, num=TEST_END-TEST_START)
+expected_ts = TimeSeries(TEST_ARRAY, t0=TEST_START, dt=1, channel=TEST_CHANNEL)
+expected_ts.write(TEMP, format='gwf')
+
+
+# # -- unit tests -------------------------------------------------------------
+
+def test_read():
+    # test reading primary TimeSeries file
+    # try finally for closing temp file
+    try:
+        actual_ts = lasso.get_primary_ts(filepath=TEMP, channel=TEST_CHANNEL,
+                                         start=TEST_START, end=TEST_END,
+                                         cache=None, nproc=1)
+    finally:
+        TEMP.close
```

Comment:
If the above is transformed into a fixture, then the following changes need to be made here:

```suggestion
def test_read(expected_ts, expected_ts_file):
    actual_ts = lasso.get_primary_ts(
        filepath=expected_ts_file,
        channel=TEST_CHANNEL,
        start=TEST_START,
        end=TEST_END,
        cache=None,
        nproc=1,
    )
```

Reply:
This has been resolved.